marimo/_server/utils.py

```diff
@@ -31,32 +34,199 @@ def canonicalize_filename(filename: str) -> str:
     return os.path.expanduser(filename)
 
 
-def find_free_port(port: int, attempts: int = 100) -> int:
-    """Find a free port or move to the next one recursively"""
+_DEFAULT_BACKLOG = 128
 
-    import socket
+
+# From Tornado (Apache 2.0)
+# https://github.com/tornadoweb/tornado/blob/8a953697888d463f48090d500268892a7384e6b1/tornado/netutil.py#L56
```

Comment:
Worth refactoring into a util file? Now this is over half the file

Reply:
I'm generally not a fan of creating tons of files. This file is still only 290 lines long.